updater/lib/dependabot/file_fetcher_command.rb

```diff
@@ -7,16 +7,20 @@
 require "dependabot/opentelemetry"
 require "dependabot/updater"
 require "octokit"
+require "sorbet-runtime"
 
 module Dependabot
   class FileFetcherCommand < BaseCommand
+    extend T::Sig
     # BaseCommand does not implement this method, so we should expose
     # the instance variable for error handling to avoid raising a
     # NotImplementedError if it is referenced
+    sig { returns(T.nilable(String)) }
     attr_reader :base_commit_sha
 
-    def perform_job # rubocop:disable Metrics/PerceivedComplexity,Metrics/AbcSize
-      @base_commit_sha = nil
+    sig { returns(T.nilable(Integer)) }
+    def perform_job # rubocop:disable Metrics/PerceivedComplexity,Metrics/AbcSize,Metrics/MethodLength
```

Comment:
nit, I know it is unrelated to the sorbet changes, but we could extract some methods to allow us to remove these offenses rather than disabling more cops.